pvlib/iotools/psm3.py

```diff
@@ -0,0 +1,174 @@
+             full_name=PVLIB_PYTHON, affiliation=PVLIB_PYTHON):
+    """
+    Get PSM3 data
+
+    Parameters
+    ----------
+    latitude : float or int
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float or int
+        in decimal degrees, between -180 and 180, east is positive
+    api_key : str
+        required NREL Developer Network API key
+    email : str
+        required, system uses this to automatically communicate messages back
+        to the user only if necessary
+    names : str
+        PSM3 API parameter specifing year or TMY variant to download, see notes
+        below for options, default is ``'tmy'``
+    interval : int
+        interval size in minutes, can only be either 30 or 60, default is 60
+    full_name : str
+        optional, default is "pvlib python"
+    affiliation : str
+        optional, default is "pvlib python"
+
+    Returns
+    -------
+    headers : dict
+        metadata from NREL PSM3 about the record, see notes for fields
+    data : pandas.DataFrame
+        timeseries data from NREL PSM3
+
+    Raises
+    ------
+    requests.HTTPError
+        if the request return status is not ok then the ``'errors'`` from the
+        JSON response will be returned as an exception
+
+    Notes
+    -----
+    The required NREL developer key, `api_key`, is available for free by
+    registering at the `NREL Developer Network <https://developer.nrel.gov/>`_.
+
+    .. warning:: The "DEMO_KEY" `api_key` is severely rate limited and may
+        result in rejected requests.
+
+    The PSM3 API `names` parameter must be a single value from the following
+    list::
+
+        ['1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005',
+         '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
+         '2014', '2015', '2016', '2017', 'tmy', 'tmy-2016', 'tmy-2017',
+         'tdy-2017', 'tgy-2017']
+
+    The return is a tuple with two items. The first item is a header with
+    metadata from NREL PSM3 about the record containing the following fields:
+
+    * Source
+    * Location ID
+    * City
+    * State
+    * Country
+    * Latitude
+    * Longitude
+    * Time Zone
+    * Elevation
+    * Local Time Zone
+    * Dew Point Units
+    * DHI Units
+    * DNI Units
+    * GHI Units
+    * Temperature Units
+    * Pressure Units
+    * Wind Direction Units
+    * Wind Speed
+    * Surface Albedo Units
+    * Version
+
+    The second item is a dataframe with the timeseries data downloaded.
+
+    See Also
+    --------
+    pvlib.iotools.read_tmy2, pvlib.iotools.read_tmy3
+
+    References
+    ----------
+
+    * `NREL Developer Network - Physical Solar Model (PSM) v3
+      <https://developer.nrel.gov/docs/solar/nsrdb/psm3_data_download/>`_
+    * `NREL National Solar Radiation Database (NSRDB)
+      <https://nsrdb.nrel.gov/>`_
+
+    """
+    longitude = ('%9.4f' % longitude).strip()
```

Comment:
If the formatting is important, we need a comment saying so. If not, let's remove it.

Reply:
I disagree, the WKT is unnecessarily strict, I think we can do this nice thing for the user and make it just work. The user inputs two numbers, either integer or float, and we convert it into a WKT which has 4 decimals, longitude before latitude, and exactly one space between them. I think we could possibly add a `format_WKT()` method in tools, instead of embedding it here, so that other tools could use that method, but asking the user to format the WKT is IMHO not a wise choice.

Reply:
maybe these lines need some comments to make it clearer to future developers?

Reply:
I think using a helper function that returns the formatted string would help a lot

Reply:
All I'm asking for is a code comment along the lines of `# format into WKT spec`